evm/evm_client.go

```diff
@@ -217,57 +188,50 @@ func (ec *Client) SubmitDataRootTupleRoot(
 		sigs,
 	)
 	if err != nil {
-		return err
-	}
-
-	if !waitToBeMined {
-		return nil
-	}
-
-	ec.logger.Debug("waiting for data commitment to be confirmed", "nonce", newNonce, "hash", tx.Hash().String())
-	receipt, err := ec.waitForTransaction(ctx, tx)
-	if err == nil && receipt != nil && receipt.Status == 1 {
-		ec.logger.Info("relayed data commitment", "nonce", newNonce, "hash", tx.Hash().String())
-		return nil
+		return nil, err
 	}
-	ec.logger.Error("failed to relay data commitment", "nonce", newNonce, "hash", tx.Hash().String())
-	return err
+	return tx, nil
 }
 
-func (ec *Client) NewTransactOpts(ctx context.Context, gasLim uint64) (*bind.TransactOpts, error) {
-	builder := newTransactOptsBuilder(ec.privateKey)
+// NewTransactionOpts creates a new transaction Opts to be used when submitting transactions.
+func (ec *Client) NewTransactionOpts(ctx context.Context) (*bind.TransactOpts, error) {
+	builder := newTransactOptsBuilder(ec.PrivateKey)
 
-	ethClient, err := ethclient.Dial(ec.evmRPC)
+	ethClient, err := ethclient.Dial(ec.EvmRPC)
 	if err != nil {
 		return nil, err
 	}
 
-	opts, err := builder(ctx, ethClient, gasLim)
+	opts, err := builder(ctx, ethClient, ec.GasLimit)
 	if err != nil {
 		return nil, err
 	}
 	return opts, nil
 }
 
 func (ec *Client) StateLastEventNonce(opts *bind.CallOpts) (uint64, error) {
-	nonce, err := ec.wrapper.StateEventNonce(opts)
+	nonce, err := ec.Wrapper.StateEventNonce(opts)
 	if err != nil {
 		return 0, err
 	}
 	return nonce.Uint64(), nil
 }
 
-func (ec *Client) waitForTransaction(
+func (ec *Client) WaitForTransaction(
 	ctx context.Context,
+	backend bind.DeployBackend,
 	tx *coregethtypes.Transaction,
 ) (*coregethtypes.Receipt, error) {
-	ethClient, err := ethclient.Dial(ec.evmRPC)
-	if err != nil {
-		return nil, err
+	ec.logger.Debug("waiting for transaction to be confirmed", "hash", tx.Hash().String())
+
+	receipt, err := bind.WaitMined(ctx, backend, tx)
```

Comment:
[blocking question]
do we know if we should still be using `WaitMined` given eth is no longer PoW?

Reply:
I checked the repo and no changes have been done to it. Probably they should rename it but they still didn't.
Anyway, it's not related to the underlying consensus, it just keeps polling if the transaction was included in some block.